### ISSUE TITLE
bug(skeval/src): train() permanently mutates instance hyperparameters as a side effect

### ISSUE THREAD

**Issue by direkkakkar319-ops** [skeval-ai/skeval#114]
`src/skeval/classifier/sentence_classifier.py` mutates `self.epochs`, `self.batch_size`, and `self.lr` when overrides are passed to `train()`.

```python
# current — permanently overwrites instance state
if epochs is not None:
    self.epochs = epochs
if batch_size is not None:
    self.batch_size = batch_size
if lr is not None:
    self.lr = lr
```

Calling `clf.train(sentences, labels, epochs=5)` on a classifier initialised with `epochs=10` permanently changes `clf.epochs` to `5`. Any subsequent call to `fit()` will silently use `epochs=5`. This contradicts how sklearn-compatible estimators are expected to behave - hyperparameters are set in `__init__` and not mutated at call time.

**Fix:** Use local variables inside `train()` instead of mutating `self`.

```python
_epochs = epochs if epochs is not None else self.epochs
_batch_size = batch_size if batch_size is not None else self.batch_size
_lr = lr if lr is not None else self.lr
```

**Comment by Storm1289** [comment]
Hi @direkkakkar319-ops , can I work on this

**Comment by direkkakkar319-ops** [comment]
@Storm1289 yes go for it
